services/horizon/internal/render/sse/main.go

```diff
@@ -132,6 +133,55 @@ func getJSON(val interface{}) string {
 	return string(js)
 }
 
+// Support notifying open SSE requests via channel based on related changes done
+// in the History database. Channel also includes reference counting for safe removal.
+type PubsubChannel struct {
+	channel        chan struct{}
+	referenceCount int
+}
+
+var pubsub_channels = make(map[string]PubsubChannel)
```

Comment:
maps are not safe for concurrent use. you need to use write/read locks using the sync package